File: planet-express/java/src/Crew.java

public class Crew {
	int thirst = 50;
	int work = 50;
	int horde = 50;
	int hunger = 50;
	int receipts = 50;
	boolean stable = true;
	int score = 50;
	
	public void drink() {
		thirst -= 5;
		work += 5;
	}

	public void deliver() {
        work -= 5;
        receipts += 5;
    }

    public void steal() {
        horde += 5;
        work += 5;
    }

    public void eat() {
        hunger -= 5;
        work += 5;
    }

    public void account() {
        receipts -= 5;
        horde -= 5;
    }

    public int score() {
        if (thirst >= 100 || work >= 100 || horde <= 0 || hunger >= 100 || receipts <= 0) {
            score = 0;
        }
        return score;
    }

    public boolean check() {
        if (score <= 0) {
            stable = false;
            return false;
        }
        else {
            if (thirst >= 85) {
                System.out.println("Fry is scary-thirsty!");
            }

            if (work >= 85) {
                System.out.println("Leela is exhausted from all this work!");
            }

            if (horde <= 15) {
                System.out.println("Bender is sad because he doesn't have enough treasure!");
            }

            if (hunger >= 85) {
                System.out.println("Zoidberg is going to eat something and it's not going to be pretty!");
            }

            if (receipts <= 15) {
                System.out.println("Hermes doesn't have enough receipts!");
            }

            return true;
        }

    }

    public static void main (String[] args) {
        Crew spaceship = new Crew();
        System.out.println(spaceship.check());
        System.out.println(spaceship.work);
        for (int i = 0; i < 7; i++) {
            spaceship.drink();
        }
        System.out.println(spaceship.work);
        System.out.println(spaceship.check());
        for (int j = 0; j < 3; j++) {
            spaceship.drink();
        }
        System.out.println(spaceship.work);
        System.out.println(spaceship.score());
        System.out.println(spaceship.check());
    }
}

File: planet-express/ruby/crew.rb
class Crew
  attr_accessor :thirst, :work, :horde, :hunger, :receipts, :stable

  def initialize
    @thirst = 50
    @work = 50
    @horde = 50
    @hunger = 50
    @receipts = 50
    @stable = true
  end

  def drink
    @thirst -= 5
    @work += 5
  end

  def deliver
    @work -= 5
    @receipts += 5
  end

  def steal
    @horde += 5
    @work += 5
  end

  def eat
    @hunger -= 5
    @work += 5
  end

  def account
    @receipts -= 5
    @horde -= 5
  end

  def score
  end

  def check
    # if score is too low, end game (need to determine proper score)
    if score < 100
      @stable = false
      return false
    else
      # check to see if you need to print warnings
      if @thirst >= 85
        puts "Fry is scary-thirsty!"
      end

      if @work >= 85
        puts "Leela is exhausted from all this work!"
      end

      if @horde <= 15
        puts "Bender is sad because he doesn't have enough treasure!"
      end

      if @hunger >= 85
        puts "Zoidberg is going to eat something and it's not going to be pretty!"
      end

      if @receipts <= 15
        puts "Hermes doesn't have enough receipts!"
      end

      return true
    end
  end

end
